cli/pcluster/config/validators.py

```diff
@@ -593,7 +593,36 @@ def url_validator(param_key, param_value, pcluster_config):
             bucket, key = match.group(1), match.group(2)
             boto3.client("s3").head_object(Bucket=bucket, Key=key)
         except ClientError:
-            warnings.append("The S3 object does not exist or you do not have access to it.")
+
+            # custom_chef_cookbook accepts S3 URIs. Check that the referenced bucket
+            # is in s3_read_resource or s3_read_write_resource.
+            if param_key == "custom_chef_cookbook":
+                cluster_section = pcluster_config.get_section("cluster")
+                s3_read_resource = cluster_section.get_param_value("s3_read_resource")
+                s3_read_write_resource = cluster_section.get_param_value("s3_read_write_resource")
+
+                if s3_read_resource == "*" or s3_read_write_resource == "*":
+                    pass
+                else:
+                    match_bucket_from_arn = r"(?<=arn:aws:s3:::)([^*/]*)"
+                    s3_read_bucket = (
+                        re.search(match_bucket_from_arn, s3_read_resource).group(0) if s3_read_resource else None
+                    )
+                    s3_write_bucket = (
+                        re.search(match_bucket_from_arn, s3_read_write_resource).group(0)
+                        if s3_read_write_resource
+                        else None
+                    )
+
+                    if bucket not in [s3_read_bucket, s3_write_bucket]:
+                        warnings.append(
+                            "The S3 object does not exist, or you do not have access it. Consider adding the {0} bucket in either s3_read_resource or s3_read_write_resource.".format(
```

Comment:
`"The S3 object does not exist, or you do not have access it. Consider adding the {0} bucket in either s3_read_resource or s3_read_write_resource."` --> `"The S3 object does not exist, or you do not have access it. Please be sure the cluster nodes have access to it."`  

I'd use a more generic message because there are different ways to achieve the same result. E.g. a custom IAM policy for the instances.